docs/csharp/whats-new/csharp-6.md

```diff
@@ -98,483 +36,157 @@ public class Student
-
-With C# 6, you can also await in catch expressions. This is most often
-used with logging scenarios:
+C# 5 had several limitations around where you could place `await` expressions. You can now use `await` in `catch` or `finally` expressions.  With C# 6, you can also await in catch expressions. This is most often used with logging scenarios:
 
 [!code-csharp[AwaitFinally](../../../samples/snippets/csharp/new-in-6/NetworkClient.cs#AwaitFinally)]
 
-The implementation details for adding `await` support inside `catch`
-and `finally` clauses ensures that the behavior is consistent with the
-behavior for synchronous code. When code executed in a `catch` or `finally`
-clause throws, execution looks for a suitable `catch` clause in the next
-surrounding block. If there was a current exception, that exception is
-lost. The same happens with awaited expressions in `catch` and `finally`
-clauses: a suitable `catch` is searched for, and the current exception,
-if any, is lost.  
+The implementation details for adding `await` support inside `catch` and `finally` clauses ensure that the behavior is consistent with the behavior for synchronous code. When code executed in a `catch` or `finally` clause throws, execution looks for a suitable `catch` clause in the next surrounding block. If there was a current exception, that exception is lost. The same happens with awaited expressions in `catch` and `finally` clauses: a suitable `catch` is searched for, and the current exception, if any, is lost.  
 
 > [!NOTE]
-> This behavior is the reason it's recommended to write `catch` and `finally`
-> clauses carefully, to avoid introducing new exceptions.
+> This behavior is the reason it's recommended to write `catch` and `finally` clauses carefully, to avoid introducing new exceptions.
 
 ## Index initializers
 
-*Index Initializers* is one of two features that make collection
-initializers more consistent with index usage. In earlier releases of C#, you could use
-*collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
+*Index Initializers* is one of two features that make collection initializers more consistent with index usage. In earlier releases of C#, you could use *collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
 
 [!code-csharp[ListInitializer](../../../samples/snippets/csharp/new-in-6/initializers.cs#ListInitializer)]
 
-Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and similar types. The new syntax supports assignment using an index into the collection:
+Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and other types where the accessible `Add` method accepts more than one argument. The new syntax supports assignment using an index into the collection:
```

Comment:
I think, "similar types" here means "types with indexers". The following code:

```csharp
using System;
using System.Collections;

class Program
{
    static void Main(string[] args)
    {
        var f = new Foo
            {
                { "foo", "bar" },
                { 1, '1', "1" },
                { 2, '4', "8" },
                { 3, '9', "27" }
            };
    }
}

class Foo : IEnumerable
{
    public void Add(string s1, string s2) { Console.WriteLine("Two args"); }
    public void Add(int x, char y, string z) { Console.WriteLine("Three args"); }
    IEnumerator IEnumerable.GetEnumerator() { throw new NotImplementedException(); }
}
```

compiles and runs with the "C# 5" as the selected language version. Thanks to @drewjcooper for [the example](https://github.com/dotnet/docs/issues/5169#issuecomment-391895482). (I've updated it not to use C# 6 features.)

So, the support of the `Add` methods with more than one argument is not a new feature of C# 6. As far as I can tell from the changes in the [Collection initializers](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#collection-initializers) section between C# 5 and C# 6, the only change is that the `Add` method might be an extension method (but it's already covered in the next section of this document).

The citation from the C# 5 spec (page 157):
>  For each specified element in order, the collection initializer invokes an `Add` method on the target object with the expression list of the element initializer as argument list,...

The element initializer might be an expression list (as `{ "foo", "bar" }` in the example), which is given as an argument list to an `Add` method.

Reply:
If this is index initialization, does that rely on `Add`?

Reply:
No, object initializers using the indexer do not also require an `Add` method.

Reply:
@BillWagner now, I understand better what confuses me in the following sentence:

>Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and other types where the accessible `Add` method accepts more than one argument.

First, I was wrong about what is behind "them" in "...you can use them with...". I thought "them" refers to index initializers. However, it refers to "braces around key and value pairs" from the paragraph that is above the code block. Is that correct understanding? I don't know what others would see behind "them", though. So, maybe no changes here.

Second, the very start of the sentence might be misleading:
>Now, you can...

Given that the topic is about what's new, I've read the sentence in the following way:
> Starting with C# 6, you can...

Starting with "now" might imply that that was not possible before.

I think the "now" might be dropped.

What do you think of the following version?
>You can use them with <xref:System.Collections.Generic.Dictionary%602> collections and other types where the accessible `Add` method accepts more than one argument.

I don't know how to replace "them" (if necessary), but dropping "now" wouldn't make the sentence worse.